frontend/src/entry-points/shop.tsx

```diff
@@ -176,124 +164,97 @@ const Shop: Component = () => {
-      console.log('Received Update')
-      const app_infos = to_app_infos_by_id(resp.payload.app_infos.map((app_info) => {
-        return { ...app_info, state: AppState.Initial }
-      }))
-
-      if (isEmpty(appInfo)) {
-        // initially write the newest update to state
-        setAppInfo(app_infos)
-        db.insertMultiple(Object.values(app_infos))
-      }
-      else {
-        // all but the first update only overwrite existing properties
-        console.log('Reconceiling updates')
-        setAppInfo(produce((s) => {
-          for (const key in app_infos) {
-            const num_key = Number(key)
-            if (s[num_key] === undefined) {
-              s[num_key] = app_infos[num_key]
-            }
-            else {
-              s[num_key] = Object.assign(s[num_key], app_infos[num_key])
-            }
-          }
-        }))
-        db.updateMultiple(resp.payload.app_infos)
-      }
-
-      setlastUpdateSerial(resp.payload.serial)
-      setIsUpdating(false)
-      setlastUpdate(new Date())
-    }
-    else if (isDownloadResponseOkay(resp.payload)) {
-      const file = { base64: resp.payload.data, name: `${resp.payload.name}.xdc` }
-      db.add_webxdc(file, resp.payload.id)
-      db.update({ ...appInfo[resp.payload.id], state: AppState.Received })
-      setAppInfo(resp.payload.id, 'state', AppState.Received)
-    }
-    else if (isDownloadResponseError(resp.payload)) {
-      // @ts-expect-error waduheck
-      setAppInfo(resp.payload.id, 'state', AppState.DownloadCancelled)
-    }
-    else if (isOutdatedResponse(resp.payload)) {
-      console.log('Current version is outdated')
-      setUpdateNeeded(true)
-    }
-    else if (isUpdateSendResponse(resp.payload)) {
-      setUpdateReceived(true)
-    }
   }, lastSerial())
 
   async function handleUpdate() {
     setIsUpdating(true)
+
+    const cached_apps = cached().map(app_info => ([app_info.app_id, app_info.version] as [string, number]))
     window.webxdc.sendUpdate({
-      payload: { Update: { serial: lastUpdateSerial() } } as ShopRequest,
+      payload: { Update: { serial: lastUpdateSerial(), apps: cached_apps } } as ShopRequest,
     }, '')
   }
 
-  async function handleDownload(app_id: number) {
-    setAppInfo(Number(app_id), 'state', AppState.Downloading)
-    db.update({ ...appInfo[app_id], state: AppState.Downloading })
+  async function handleDownload(app_id: string) {
+    setAppInfo(app_id, 'state', AppState.Downloading)
     window.webxdc.sendUpdate({
       payload: { Download: { app_id } } as ShopRequest,
     }, '')
   }
 
-  async function handleForward(app_id: number) {
+  async function handleForward(app_id: string) {
     const file = await db.get_webxdc(app_id)
     if (file === undefined) {
       throw new Error('No cached file found')
     }
     window.webxdc.sendToChat({ file })
   }
 
+  async function handleRemove(app_id: string) {
+    setAppInfo(app_id, 'state', AppState.Initial)
+    db.remove_webxdc(app_id)
+  }
```

Comment:
You can see it being called from the AppList which is created in the functions jxs (around line 256)

Reply:
Yes. For webxdc I understand the rewording, but in this case it should be clear to every developer who wants to work on this code what the meaning of it is. I could add a short comment to it bot otherwise I like it better to have short names - personally.